here-naksha-app-service/src/test/java/com/here/naksha/app/service/DefaultViewHandlerTest.java

```diff
@@ -342,4 +343,84 @@ void tc5023_searchByBBox_PullingByIdWhenOutsideBBox() throws Exception {
                 .hasStreamIdHeader(streamId)
                 .hasJsonBody(expectedBodyPart, "Create Feature response body doesn't match");
     }
+
+    @Test
+    void tc5030_patchFeature_featureAvailableInAllLayer() throws Exception {
+
+        //given Feature is available in all three spaces
+        final String bodyJson = loadFileOrFail("DefaultViewHandler/TC5030_patchFeature/patch_feature.json");
+        final String expectedBodyPart = loadFileOrFail("DefaultViewHandler/TC5030_patchFeature/feature_response_part.json");
+        final String idsQueryParam = "id=FeatId-patchOp-5030";
+        String viewStreamId = UUID.randomUUID().toString();
+        String streamId = UUID.randomUUID().toString();
+
+        // when: Sent patch feature request to view space.
+        HttpResponse<String> viewResponse = getNakshaClient().post("hub/spaces/" + SPACE_ID + "/features", bodyJson, viewStreamId);
+        HttpResponse<String> spaceSearchResponse = getNakshaClient().get("hub/spaces/" + DELTA_CONFIGURED_SPACE + "/features?" + idsQueryParam, streamId);
+
+        // Then: Assert that feature was correctly patched by view space and was updated in correct space(top one -> delta)
+        assertThat(viewResponse)
+                .hasStatus(200)
+                .hasStreamIdHeader(viewStreamId)
+                .hasJsonBody(expectedBodyPart, "Patch Feature response body doesn't match")
+                .hasUuids();
```

Comment:
nth: I would change the order and run assertions run after the action (so run the `viewResponse` assertion before calling get on delta)

Reply:
this thought applies to other places - it is not critical though